### DISCUSSION TITLE
How to use stream results?

### DISCUSSION THREAD

**Discussion by KuzenkovAG** [long2ice/asynch#100] [General]
Dear Sirs,

Could you add some examples how to use stream result for cursor?

if i use like from issue example https://github.com/long2ice/asynch/issues/24:
```
    conn = await connect(
        host=settings.CLICKHOUSE_BACKEND_HOST,
        port=settings.CLICKHOUSE_BACKEND_TCP_PORT,
        user=settings.CLICKHOUSE_BACKEND_USER,
        password=settings.CLICKHOUSE_BACKEND_PASSWORD,
        database="db",
    )

    async with conn.cursor(cursor=DictCursor) as cursor:
        cursor.set_stream_results(stream_results=True, max_row_buffer=1000)
        await cursor.execute(
            """
                SELECT *
                FROM ontime
        """
        )
        for row in cursor.fetchall():
            print(row)
```
i have error:
```
RuntimeWarning: coroutine 'DictCursor.fetchall' was never awaited
  for row in cursor.fetchall():

```

**Comment by stankudrow** [comment]
@KuzenkovAG , the `cursor.fetchall()` method is a coroutine function, so it must be awaited:
```
for row in await cursor.fetchall():
    print(row)
```

When trying this approach, the cursor gets stick in the infinite loop situation (please confirm):
```
async for row in cursor:
    print(row)
``` 

The second case was the reason for the PR #112, please review, test and approve it when all things will get done.

Thank you.

**Reply by KuzenkovAG** [reply]
@stankudrow 
Kindly thanks you for your PR!

**Reply by stankudrow** [reply]
@KuzenkovAG , could you mark this as answered if it is so?
